src/layers/legacy/medCoreLegacy/gui/commonWidgets/medStatusBar.cpp

```diff
@@ -14,18 +14,25 @@
 #include <medStatusBar.h>
 
 #include <QtCore>
+#include <mutex>
 
 class medStatusBarPrivate
 {
 public:
     QBoxLayout * statusBarLayout;
+    int availableSpace; //available space on the status bar
+    int statusBarWidth;
+    std::mutex listsofMsgMutex;      // mutex to guard conflict between external signals and/or internal update
+    QList<medMessage *> hiddenMessageList; // storing messages waiting to be displayed
+    QList<medMessage *> messageList;       // storing displayed messages
 };
 
 medStatusBar::medStatusBar ( QWidget* parent ) : QStatusBar ( parent ), d ( new medStatusBarPrivate )
 {
     d->statusBarLayout = NULL;
-    this->statusBarWidth = this->size().width();
-    this->availableSpace = -1; //value before initialization (after, always positive)
+    d->statusBarWidth = this->size().width();
+    d->availableSpace = -1; //value before initialization (after, always positive)
+    d->listsofMsgMutex;
```

Comment:
Heu ca sert a rien cette ligne la non ?

Reply:
ha oui carrément c'était juste pour un point d’arrêt durant la mise au point, je vais le supprimer.

Reply:
You forgot to remove this line ^^

Reply:
done